pkg/workloads/cortex/lib/type/predictor.py

```diff
@@ -20,36 +20,67 @@
 
 from cortex.lib.log import refresh_logger, cx_logger
 from cortex.lib.exceptions import CortexException, UserException, UserRuntimeException
+from cortex.lib.type.model import Model, get_model_signature_map
+from cortex import consts
 
 
 class Predictor:
-    def __init__(self, provider, cache_dir, **kwargs):
+    def __init__(self, provider, model_dir, cache_dir, **kwargs):
         self.provider = provider
         self.type = kwargs["type"]
         self.path = kwargs["path"]
-        self.model = kwargs.get("model")
         self.python_path = kwargs.get("python_path")
         self.config = kwargs.get("config", {})
         self.env = kwargs.get("env")
-        self.signature_key = kwargs.get("signature_key")
+
+        self.model_dir = model_dir
+        self.models = []
+        if kwargs.get("models"):
+            for model in kwargs.get("models"):
+                self.models += [
+                    Model(
+                        name=model.get("name"),
+                        model=model.get("model"),
+                        signature_key=model.get("signature_key"),
+                        base_path=self._compute_model_basepath(
+                            model.get("model"), model.get("name")
+                        ),
+                    )
+                ]
 
         self.cache_dir = cache_dir
 
-    def initialize_client(self, model_dir=None, tf_serving_host=None, tf_serving_port=None):
+    def initialize_client(self, tf_serving_host=None, tf_serving_port=None):
+        signature_message = None
         if self.type == "onnx":
             from cortex.lib.client.onnx import ONNXClient
 
-            model_path = os.path.join(model_dir, os.path.basename(self.model))
-            client = ONNXClient(model_path)
-            cx_logger().info("ONNX model signature: {}".format(client.input_signature))
+            client = ONNXClient(self.models)
+            if self.models[0].name == consts.SINGLE_MODEL_NAME:
+                signature_message = "ONNX model signature: {}".format(self.models[0].signature_key)
+            else:
+                signature_message = "ONNX model signatures: {}".format(
+                    get_model_signature_map(self.models)
```

Comment:
See above. This should be `client.input_signatures` instead of `get_model_signature_map(self.models)`